lib-multisrc/madara/src/eu/kanade/tachiyomi/multisrc/madara/Madara.kt

```diff
@@ -702,42 +702,32 @@ abstract class Madara(
                 }
             }
             val genres = select(mangaDetailsSelectorGenre)
-                .map { element -> element.text().lowercase(Locale.ROOT) }
+                .map { element -> element.text() }
                 .toMutableSet()
 
             if (mangaDetailsSelectorTag.isNotEmpty()) {
                 select(mangaDetailsSelectorTag).forEach { element ->
-                    if (genres.contains(element.text()).not() &&
+                    if (genres.any { sv -> sv.lowercase(Locale.ROOT) == element.text().lowercase(Locale.ROOT) }.not() &&
                         element.text().length <= 25 &&
                         element.text().contains("read", true).not() &&
                         element.text().contains(name, true).not() &&
                         element.text().contains(name.replace(" ", ""), true).not() &&
                         element.text().contains(manga.title, true).not() &&
                         element.text().contains(altName, true).not()
                     ) {
-                        genres.add(element.text().lowercase(Locale.ROOT))
+                        genres.add(element.text())
                     }
                 }
             }
 
             // add manga/manhwa/manhua thinggy to genre
             document.selectFirst(seriesTypeSelector)?.ownText()?.let {
-                if (it.isEmpty().not() && it.notUpdating() && it != "-" && genres.contains(it).not()) {
-                    genres.add(it.lowercase(Locale.ROOT))
+                if (it.isEmpty().not() && it.notUpdating() && it != "-" && genres.any { sv -> sv.lowercase(Locale.ROOT) == it.lowercase(Locale.ROOT) }.not()) {
+                    genres.add(it)
                 }
             }
 
-            manga.genre = genres.toList().joinToString { genre ->
-                genre.replaceFirstChar {
-                    if (it.isLowerCase()) {
-                        it.titlecase(
-                            Locale.ROOT,
-                        )
-                    } else {
-                        it.toString()
-                    }
-                }
-            }
+            manga.genre = genres.toList().joinToString()
```

Comment:
```suggestion
            manga.genre = genres.distinctBy(String::lowercase).joinToString()
```

You'll also need to remove the `toMutableSet()` call and the `any` comparisons.